python/tests/expressions/test_visitors.py

```diff
@@ -827,24 +834,23 @@ def _to_manifest_file(*partitions: PartitionFieldSummary) -> ManifestFile:
     )
 
 
-def _create_manifest_evaluator(bound_expr: BoundPredicate) -> _ManifestEvalVisitor:
-    """For testing. Creates a bogus evaluator, and then replaces the expression"""
-    evaluator = _ManifestEvalVisitor(
-        Schema(NestedField(1, "id", LongType())), EqualTo(term=Reference("id"), literal=literal("foo"))
-    )
-    evaluator.partition_filter = bound_expr
-    return evaluator
+@pytest.fixture
+def string_schema() -> Schema:
```

Comment:
You may actually want to just base the tests here on the Java version, which has pretty through tests and reuses the manifest metadata: https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/expressions/TestInclusiveManifestEvaluator.java

Reply:
Added all of them: https://github.com/apache/iceberg/pull/6140/commits/8b3f2696291f8bdfa8007bbd55c6f3738a4182e6
With one minor change, I made the field IDs consecutive 🙀:
https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/expressions/TestInclusiveManifestEvaluator.java#L57-L58